mqtt/mqtt-bridge/src/persist/persistor.rs

```diff
@@ -66,7 +54,7 @@ impl<'a> Persist<'a> for InMemoryPersist {
         state_lock.remove_in_flight(&key)
     }
 
-    async fn loader(&'a mut self) -> Arc<Mutex<InMemoryMessageLoader>> {
+    async fn loader(&mut self) -> Arc<Mutex<MessageLoader<S>>> {
         Arc::clone(&self.loader)
```

Comment:
Shouldn't we create loader in this methods instead of constructor?

Reply:
We discussed this in the last PR. The problem with this is that it allow multiple loaders to exist, which the waking logic is not set up to deal with.